lib/storage/keyval/operations.go

```diff
@@ -66,6 +74,14 @@ func (b *backend) GetSiteOperation(siteDomain, operationID string) (*storage.Sit
 	}
 	utils.UTC(&op.Created)
 	utils.UTC(&op.Updated)
+
+	// Operations that are not expected to change in the future are the only operations that are safe to cache
+	if op.State == storage.OperationPhaseStateCompleted || op.State == storage.OperationPhaseStateRolledBack {
```

Comment:
This should be comparing with the operation states (e.g. `ops.SiteOperationStateCompleted`), not operation phase states. Also, the operation cannot be "rolled back" (it's an individual phase state). The only terminal operation state is `ops.SiteOperationStateCompletes` IIRC (since if it's "failed", it can still become "completed").